central/sensor/service/pipeline/nodeinventory/pipeline.go

```diff
@@ -92,10 +93,10 @@ func (p *pipelineImpl) Run(ctx context.Context, _ string, msg *central.MsgFromSe
 		return errors.WithMessagef(err, "node does not exist: %s", ninv.GetNodeId())
 	}
 
-	// Discard message if NodeScanning v4 and v2 are running in parallel - v4 scans are prioritized in that case.
-	// The message will be kept even if a v4 scan is already persisted for the node if the feature flag for
-	// Scanner v4 has been disabled. This ensures node scans are updated even if a customer falls back to Scanner v2.
-	if node.GetScan() != nil && node.GetScan().GetScannerVersion() == storage.NodeScan_SCANNER_V4 && features.ScannerV4.Enabled() {
```

Comment:
Could you double-check that we don't need the `node.GetScan() != nil` later?
It is not needed in this func but I am not sure about `sendComplianceAck`.